src/AutoRest.CSharp/Mgmt/Output/MgmtLongRunningOperation.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Transactions;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace AutoRest.CSharp.Mgmt.Output
+{
+    internal class MgmtLongRunningOperation : TypeProvider
+    {
+        public bool IsGeneric { get; }
+
+        protected override string DefaultName { get; }
+
+        protected override string DefaultAccessibility => "internal";
+
+        public MgmtLongRunningOperation(bool isGeneric) : base(MgmtContext.Context)
+        {
+            IsGeneric = isGeneric;
+            DefaultName = $"{MgmtContext.Context.DefaultNamespace.Split('.').Last()}ArmOperation";
+            BaseType = isGeneric ? typeof(ArmOperation<>) : typeof(ArmOperation);
+            WaitMethod = isGeneric ? "WaitForCompletionAsync" : "WaitForCompletionResponseAsync";
+            OperationType = isGeneric ? typeof(ArmOperation<>) : typeof(ArmOperation);
+            ResponseType = isGeneric ? typeof(Response<>) : typeof(Response);
+            OperationOrResponseType = isGeneric ? typeof(OperationOrResponseInternals<>) : typeof(OperationOrResponseInternals);
+            OperationSourceString = isGeneric ? (FormattableString)$"{typeof(IOperationSource<>)} source, " : (FormattableString)$"";
```

Comment:
Is it used anywhere?

Reply:
Well, no, this is introduced during the refactor. I will remove this shortly